src/NServiceBus.Newtonsoft.Json/JsonMessageSerializer.cs

```diff
@@ -26,10 +27,11 @@ class JsonMessageSerializer : IMessageSerializer
         {
             this.messageMapper = messageMapper;
 
-            settings ??= new JsonSerializerSettings
```

Comment:
Would it be better to still use the default ctor and then check `settings.TypeNameHandling == TypeNameHandling.Auto`? The benefit of this would be that if at any time they change the default in newtonsoft we would still be using the default setting (whatever that is at the point in time).

I'm mentioning it because at this stage our implemention is:

When no setting provided use the default settings of the version we bound

your PR would change that though to

Whe no setting provided use the default setting exception set the `TypeNameHandling` to always be `Auto`.

Reply:
> The benefit of this would be that if at any time they change the default in newtonsoft we would still be using the default setting (whatever that is at the point in time).

I'm pretty sure they will never change that default in Json.NET and also we don't want to adopt changing defaults automatically either as that would be compatiblity-breaking change.

> When no setting provided use the default settings of the version we bound

not sure what you mean by that? The current implementation is "use the default settings but set TypeNameHandling to Auto".

Reply:
My point was that the current design of the newtonsoft implementation is that it takes the package version default behavior of the default constructor of the JsonSettings. The change here does multiple things:

- It changes that design
- It introduces backward compatibility by hardcoding the auto-handling
- It logs a warning

That's why I brought this up to clarify if we really should couple those things together in one go. As long as we have thought about the implications, I'm fine. I was just thinking that if we want to not change the previous design assumptions, staying with newing up the default settings and logging when type name handling is set to auto would achieve the goal of making the users aware.

Reply:
OH duh, I completely missed that we set it before https://github.com/Particular/NServiceBus.Newtonsoft.Json/pull/215/files#diff-f83cd6aed4a4f832e85cf6e172f0586133dd3318b269fd9d226a9c2bbaac0f72L31 in the diff. Never mind my comment. Sorry for holding things up. I should have looked closer